### PR TITLE
U ichanges

### PR DESCRIPTION
**Pull request by abhay-mandal**

### CODE REVIEW

File: views/index.html [hunk 1]
```diff
@@ -13,38 +13,55 @@
...
-            <div class="clearfix pt-3 pb-2">
-                <div class="float-right">
+            <div class="clearfix pb-2">
                     <div class="form-group">
                         <div class="float-left ml-2">
-                            <label>Select a Bot</label>
-                            <select ng-model="botDetailsModel" ng-change="changeBotDetails()" ng-options="bot.botName for bot in botDetails track by bot.botId" class="form form-control inline-block">
-                                <option value="">Select a Bot</option>
+                            <label>Select Flow Name</label>
+                            <select ng-model="botDetailsModel" ng-change="changeBotFlowType()" ng-options="bot.botName for bot in botDetails track by bot.botId" class="form form-control inline-block">
```
Comment (amruta-jv) [inline views/index.html]: `changeBot()?`

File: views/index.html [hunk 2]
```diff
@@ -13,38 +13,55 @@
...
+                                <option value="">Select Flow Name</option>
```
Comment (amruta-jv) [inline views/index.html]: `Select bot`

File: views/index.html [hunk 3]
```diff
@@ -13,38 +13,55 @@
...
+                                    <label class="botflow-container">
+                                        <input ng-model="MessageByType" ng-change='changeMessageByType(MessageByType)' type="radio" name="options" id="option3" value=""> 
+                                        Incoming and Outgoing
+                                        <span class="checkmark"></span>
+                                    </label>
+                                </div>
+                        </div>
                         <div class="float-left ml-3">
-                            <label class="">Select Date Range</label>
+                            <!-- <label class="">Date Range</label> -->
```
Comment (amruta-jv) [inline views/index.html]: Remove commented line.

File: views/js/controller.js [hunk 4]
```diff
@@ -7,16 +7,24 @@ app.controller("myController", function ($scope, reportServiceData) {
     $scope.conversationVsUsers = false;
     $scope.messageVsUsers = false;
     $scope.usersVsFiles = false;
+    $scope.reportGrid_flag = false;
```
Comment (amruta-jv) [inline views/js/controller.js]: This can be camel casing instead underscore.

File: views/js/controller.js [hunk 5]
```diff
@@ -7,16 +7,24 @@ app.controller("myController", function ($scope, reportServiceData) {
...
 
-    $scope.date = {
+    $scope.datePicker = {
         startDate: moment().subtract(29, "days"),
         endDate: moment()
     };
 
+
+    //dafault radio flow type
+    $scope.MessageByType = ""; //both bot and user
```
Comment (amruta-jv) [inline views/js/controller.js]: Camel casing.

Comment (amruta-jv) [inline views/js/controller.js]: This can be `messageByType`.

File: views/js/controller.js [hunk 6]
```diff
@@ -120,54 +131,70 @@ app.controller("myController", function ($scope, reportServiceData) {
...
+            })
+            .catch(function (err) {
+                $scope.reportGrid_flag = false;
+                console.log(err);
+            });
+    }
+
+    
+
     function generateChartConversationVsUsers(data) {
```
Comment (amruta-jv) [inline views/js/controller.js]: generateConversationVsUsersChart()

File: views/js/controller.js [hunk 7]
```diff
@@ -188,16 +215,20 @@ app.controller("myController", function ($scope, reportServiceData) {
...
-                "balloonText": "<span style='font-size:12px;'>[[title]] :<span style='font-size:20px;'>[[value]]</span> [[additional]]</span>",
-                "fillAlphas": 1,
-                "title": "Conversations",
-                "type": "column",
+                // "alphaField": "alpha",
+                // "balloonText": "<span style='font-size:12px;'>[[title]] :<span style='font-size:20px;'>[[value]]</span> [[additional]]</span>",
+                // "fillAlphas": 1,
+                // "title": "Conversations",
+                // "type": "column",
+                // "valueField": "conversations",
```
Comment (amruta-jv) [inline views/js/controller.js]: Please remove commented code if we don't need it.

File: views/js/services.js [hunk 8]
```diff
@@ -27,19 +26,19 @@
...
             totalFilesUrl = `${totalFilesUrl}?fromDate=${fromDate}&toDate=${toDate}`; 
         
             if (botId) {
-                totalFilesUrl = `${totalFilesUrl}&botId=${botId}`;
+                totalFilesUrl = `${totalFilesUrl}&botId=${botId}&flowType=${messageBy}`;
             }
         }
         else if (botId) {
-            totalFilesUrl = `${totalFilesUrl}?botId=${botId}`;
+            totalFilesUrl = `${totalFilesUrl}?botId=${botId}&flowType=${messageBy}`;
```
Comment (amruta-jv) [inline views/js/services.js]: Rename query param.

File: views/index.html (hunk 2 above)
Comment (abhay-mandal) [reply views/index.html]: sruthy has given this label

File: views/js/controller.js (hunk 7 above)
Comment (abhay-mandal) [reply views/js/controller.js]: i may require these lines

File: views/js/services.js (hunk 8 above)
Comment (abhay-mandal) [reply views/js/services.js]: done

File: views/js/controller.js [hunk 9]
```diff
@@ -7,16 +7,24 @@ app.controller("myController", function ($scope, reportServiceData) {
     $scope.conversationVsUsers = false;
     $scope.messageVsUsers = false;
     $scope.usersVsFiles = false;
+    $scope.reportGridFlag = false;
     $scope.botDetails;
+    $scope.reportGridData;
+
+    $scope.filterType = { filterType: "menu" };
 
     var created_by = "5c2f07fd9d1f491b0d179c48"; //"5afec186bba8b35cbb98bc61";
```
Comment (amruta-jv) [inline views/js/controller.js]: Even this one, let's not have underscore in variable name.

File: views/js/controller.js [hunk 10]
```diff
@@ -57,23 +62,26 @@ app.controller("myController", function ($scope, reportServiceData) {
...
-        getTotalMessageCount(newDate,botId);
-        getConversationVsUsers(newDate,botId);
-        getMessageVsUsers(newDate,botId);
-        getUsersVsFiles(newDate,botId);
+        getTotalMessageCount(newDate,botId,messageBy);
+        getTotalfiles(newDate,botId,messageBy);
+        getConversationVsUsers(newDate,botId,messageBy);
+        getMessageVsUsers(newDate,botId,messageBy);
+        getUsersVsFiles(newDate,botId,messageBy);
+        getReportGridData(newDate,botId,messageBy);
```
Comment (amruta-jv) [inline views/js/controller.js]: It's always good to leave space after punctuation.

**Pull request merged**